### PR TITLE
Automatic unit test generator

### PR DESCRIPTION
**Pull request by perezjosibm**
Introducing an automated unit test generator for the benchmark classes in CBT.

Detailed documentation added to the docs/.

* To generate a baseline and unit test code, run
`# python3 tools/serialise_benchmark.py`

and example of the expected output is shown:
cbt_utest_gen

* To run the unit tests:
`# python3 -m pytest -p no:cacheprovider tests/`

an example of the expected output is shown below:

cbt_utests_run

### CODE REVIEW

**Comment by perezjosibm** [comment]
Hi @sseshasa would you be able to review my PR please? Many thanks in advance

**Comment by sseshasa** [comment]
> Hi @sseshasa would you be able to review my PR please? Many thanks in advance

@perezjosibm sorry for the delay, I will try and review this in the next couple of days.

**Review by sseshasa — APPROVED** [review]
LGTM. Minor nits only.

File: common.py [hunk 1]
```diff
@@ -116,17 +142,18 @@ def get_localnode(nodes):
...
+#    local_fqdn = get_fqdn_local()
+#    local_hostname = socket.gethostname()
+#    local_short_hostname = local_hostname.split('.')[0]
+#
+#    remote_host = settings.host_info(nodes_list[0])['host']
+#    #logger.debug('remote_host=%s, local_fqdn=%s local_hostname=%s local_short_hostname=%s'
+#    #                 % (remote_host, str(local_fqdn), str(local_hostname), str(local_short_hostname) ))
+#    if remote_host in ('localhost', local_fqdn, local_hostname, local_short_hostname):
+#        return remote_host
+#    return None
```
[the remark this answers is not in the document]
Comment (perezjosibm) [reply common.py:156]: Removed 👍

**Pull request merged**